Fundamentals/Setup/Upgrading/umbraco10-breaking-changes.md

```diff
@@ -0,0 +1,149 @@
+
+- Constants for SQL Server provider name moved+consolidated from _Core.Constants.DatabaseProviders_ & _Core.Constants.-DbProviderNames_ to _Umbraco.Cms.Persistence.SqlServer.Constants_
+
+- Various SQL Server related services moved from the Umbraco.Infrastructure project to the new Umbraco.Cms.
+Persistence.- SqlServer project with altered namespaces e.g.
+
+- _SqlServerSyntaxProvider_, _SqlServerBulkSqlInsertProvider_, _SqlServerDatabaseCreator_
+
+**Added various methods / properties to ISqlSyntaxProvider, these must be implemented in any downstream implementation e.g:**
+
+- _ISqlSyntaxProvider.HandleCreateTable(IDatabase,TableDefinition,Boolean)_
+- _ISqlSyntaxProvider.GetFieldNameForUpdate()_
+- _ISqlSyntaxProvider.GetColumn(DatabaseType,String,String,String,String,Boolean)_
+- _ISqlSyntaxProvider.InsertForUpdateHint(Sql)_
+- _ISqlSyntaxProvider.AppendForUpdateHint(Sql)_
+- _ISqlSyntaxProvider.LeftJoinWithNestedJoin(Sql,Func<Sql,Sql>,String)_
+
+## v10: Update to ImageSharp v2
+
+**Update dependency versions**:
+
+- _SixLabors.ImageSharp from 1.0.4 to 2.1.1 (see breaking changes in v2.0.0 release notes);_
+
+- _SixLabors.ImageSharp.Web from 1.0.5 to 2.0.0 (see breaking changed in release notes);_
+
+Renamed CachedNameLength property to CacheHashLength on ImagingCacheSettings.
+
+Moved ImageSharpImageUrlGenerator from project Umbraco.Infrastructure to Umbraco.Web.Common and updated the corresponding namespace and DI registration (from AddCoreInitialServices() to AddUmbracoImageSharp());
+
+Moved ImageSharp configuration from the _AddUmbracoImageSharp()_ extension method into separate _IConfigureOptions<>_ implementations:
+
+- _The middleware is configured in ConfigureImageSharpMiddlewareOptions (which also replaces ImageSharpConfigurationOptions that previously only set the default ImageSharp configuration);_
+
+- _The default physical cache is configured in ConfigurePhysicalFileSystemCacheOptions._
+
+## V10: Migrate member properties to columns on the member table
+
+This is breaking because it is no longer possible access the properties listed below through the _IMember.Properties_ collection, you must now access them through their specific properties I.E. _IMember.IsLockedOut_.
+
+- _umbracoMemberFailedPasswordAttempts_
+- _umbracoMemberApproved_
+- _umbracoMemberLockedOut_
+- _umbracoMemberLastLockoutDate_
+- _umbracoMemberLastLogin_
+- _umbracoMemberLastPasswordChangeDate_
+
+Additionally, when previously you resolved a member as published content, all the default properties would be there twice, for instance, _IsLockedOut_ would be there both as a property with the alias umbracoMemberLockedOut and with the alias _IsLockedOut_, now it'll only be there once, with the alias being the name of the property, so _IsLockedOut_ in this instance.
+
+Lastly the nullable dates on a user, I.E. _LastLoginLate_ will now be null instead of _DateTime.MinValue_ when getting a user with the user service.
```

Comment:
```suggestion
Lastly the nullable dates on a user, i.e. _LastLoginLate_ will now be null instead of _DateTime.MinValue_ when getting a user with the user service.
```